stream.go

```diff
@@ -48,24 +48,43 @@ func Subscribe(url, lastEventId string) (*Stream, error) {
 	return SubscribeWithRequest(lastEventId, req)
 }
 
+// SubscribeWith subscribes to the Events emitted from the specified url, using the
+func SubscribeWith(lastEventId string, client *http.Client, request *http.Request) (*Stream, error) {
+	stream := createNewStream(lastEventId, client, request)
+
+	r, err := stream.connect()
+	if err != nil {
+		return nil, err
+	}
+	go stream.stream(r)
+	return stream, nil
+}
+
 // SubscribeWithRequest will take an http.Request to setup the stream, allowing custom headers
 // to be specified, authentication to be configured, etc.
-func SubscribeWithRequest(lastEventId string, req *http.Request) (*Stream, error) {
+func SubscribeWithRequest(lastEventId string, request *http.Request) (*Stream, error) {
+	stream := createNewStream(lastEventId, http.DefaultClient, request)
+
```

Comment:
Thanks for the PR! Probably could remove the duplication and the createNewStream function by making SubscribeWithRequest call SubscribeWith using http.DefaultClient